metadata-ingestion/src/datahub/emitter/mcp_builder.py

```diff
@@ -0,0 +1,169 @@
+    )
+    wu = MetadataWorkUnit(id=f"{domain_urn}-to-{entity_urn}", mcp=mcp)
+    yield wu
+
+
+def gen_containers(
+    container_key: KeyType,
+    name: str,
+    sub_types: List[str],
+    parent_container_key: Optional[PlatformKey] = None,
+    domain_urn: Optional[str] = None,
+) -> Iterable[MetadataWorkUnit]:
+    container_urn = make_container_urn(
+        guid=container_key.guid(),
+    )
+    mcp = MetadataChangeProposalWrapper(
+        entityType="container",
+        changeType=ChangeTypeClass.UPSERT,
+        entityUrn=f"{container_urn}",
+        # entityKeyAspect=ContainerKeyClass(guid=schema_container_key.guid()),
+        aspectName="containerProperties",
+        aspect=ContainerProperties(
+            name=name, customProperties=dataclasses.asdict(container_key)
+        ),
+    )
+    wu = MetadataWorkUnit(id=f"container-info-{name}-{container_urn}", mcp=mcp)
+    yield wu
+
+    mcp = MetadataChangeProposalWrapper(
+        entityType="container",
+        changeType=ChangeTypeClass.UPSERT,
+        entityUrn=f"{container_urn}",
+        # entityKeyAspect=ContainerKeyClass(guid=schema_container_key.guid()),
+        aspectName="dataPlatformInstance",
+        aspect=DataPlatformInstance(
+            platform=f"{make_data_platform_urn(container_key.platform)}"
+        ),
+    )
+    wu = MetadataWorkUnit(
+        id=f"container-platforminstance-{name}-{container_urn}", mcp=mcp
+    )
+    yield wu
+
+    # Set subtype
+    subtype_mcp = MetadataChangeProposalWrapper(
+        entityType="container",
+        changeType=ChangeTypeClass.UPSERT,
+        entityUrn=f"{container_urn}",
+        # entityKeyAspect=ContainerKeyClass(guid=schema_container_key.guid()),
+        aspectName="subTypes",
+        aspect=SubTypesClass(typeNames=sub_types),
+    )
+    wu = MetadataWorkUnit(
+        id=f"container-subtypes-{name}-{container_urn}", mcp=subtype_mcp
+    )
+    yield wu
+
+    if domain_urn:
+        yield from add_domain_to_entity_wu(
+            entity_type="container",
+            entity_urn=container_urn,
+            domain_urn=domain_urn,
+        )
+
+    if parent_container_key:
+        parent_container_urn = make_container_urn(
+            guid=parent_container_key.guid(),
+        )
+
+        # Set database container
+        db_container_mcp = MetadataChangeProposalWrapper(
+            entityType="container",
+            changeType=ChangeTypeClass.UPSERT,
+            entityUrn=f"{container_urn}",
+            # entityKeyAspect=ContainerKeyClass(guid=schema_container_key.guid()),
+            aspectName="container",
+            aspect=ContainerClass(container=parent_container_urn),
+            # aspect=ContainerKeyClass(guid=database_container_key.guid())
+        )
+        wu = MetadataWorkUnit(
+            id=f"container-parent-container-{name}-{container_urn}-{parent_container_urn}",
+            mcp=db_container_mcp,
+        )
+
+        yield wu
+
+
+def add_dataset_to_container(
```

Comment:
We will also want to implement add_chart_to_container and add_dashboard_to_container

Reply:
(for the future)